BUILDING.md

```diff
@@ -8,21 +8,90 @@ If you consistently can reproduce a test failure, search for it in the
+* **Tier 2**: Full test coverage but more limited maintenance,
+  often provided by the vendor of the platform. 
+* **Experimental**: Known to compile but not necessarily reliably or with
+  a full passing test suite. These are often working to be promoted to Tier
+  2 but are not quite ready. There is at least one individual actively
+  providing maintenance and the team is striving to broaden quality and
+  reliability of support.
+
+### Supported platforms
+
+|  System      | Support type | Version                          | Architectures        | Notes            |
+|--------------|--------------|----------------------------------|----------------------|------------------|
+| GNU/Linux    | Tier 1       | kernel >= 2.6.18, glibc >= 2.5   | x86, x64, arm, arm64 |                  | 
+| macOS        | Tier 1       | >= 10.10                         | x64                  |                  |
+| Windows      | Tier 1       | >= Windows 7 or >= Windows2008R2 | x86, x64             |                  |
+| SmartOS      | Tier 2       | >= 15 < 16.4                     | x86, x64             | see note1        |
+| FreeBSD      | Tier 2       | >= 10                            | x64                  |                  |
+| GNU/Linux    | Tier 2       | kernel >= 4.2.0, glibc >= 2.19   | ppc64be              |                  |
+| GNU/Linux    | Tier 2       | kernel >= 3.13.0, glibc >= 2.19  | ppc64le              |                  |
+| AIX          | Tier 2       | >= 6.1 TL09                      | ppc64be              |                  |
+| GNU/Linux    | Tier 2       | kernel >= 3.10, glibc >= 2.17    | s390x                |                  |
+| macOS        | Experimental | >= 10.8 < 10.10                  | x64                  | no test coverage |
+| Linux (musl) | Experimental | musl >= 1.0                      | x64                  |                  |
+
+note1 - The gcc4.8-libs package needs to be installed, because node
+  binaries have been built with GCC 4.8, for which runtime libraries are not
+  installed by default. For these node versions, the recommended binaries
+  are the ones available in pkgsrc, not the one available from nodejs.org.
+  Note that the binaries downloaded from the pkgsrc repositories are not
+  officially supported by the Node.js project, and instead are supported
+  by Joyent. SmartOS images >= 16.4 are not supported because
+  GCC 4.8 runtime libraries are not available in their pkgsrc repository 
+
+### Supported toolchains
+
+Depending on host platform, the selection of toolchains may vary.
+
+#### Unix
+
+* GCC 4.8.5 or newer
+* Clang 3.4.1 or newer
+
+#### Windows
+
+* Building Node: Visual Studio 2015 or Visual C++ Build Tools 2015 or newer
+* Building native add-ons: Visual Studio 2013 or Visual C++ Build Tools 2015 or newer
+
+### Shared libraries & dependencies
+
+Node.js intends to support building against shared representations of
+dependencies found in the [*deps*](./deps/) directory.
```

Comment:
I'll remove for now, and we can add back in through follow PRs if appropriate after more discussion.  What I think makes sense is that once this PR lands, I can open a new one adding that back in and we can have the discussion there as opposed to blocking getting the reset of the info in.